webpack/assets/javascripts/react_app/routes/Models/ModelsPage/index.js

```diff
@@ -0,0 +1,36 @@
+import { connect } from 'react-redux';
+import { bindActionCreators, compose } from 'redux';
+
+import ModelsPage from './ModelsPage';
+import * as ModelsPageActions from './ModelsPageActions';
+
+import { getParams } from '../../../common/urlHelpers';
+import { callOnMount } from '../../../common/HOC';
+import withDataReducer from '../../common/reducerHOC/withDataReducer';
+
+import {
+  modelsAreLoading,
+  modelsMessage,
+  modelsHaveError,
+  modelsHaveData
+} from './ModelsPageSelectors';
+
+export const reducers = { modelsPage: withDataReducer('MODELS_PAGE') };
+
+const mapDispatchToProps = dispatch =>
+  bindActionCreators(ModelsPageActions, dispatch);
+
+const mapStateToProps = state => {
+  return {
+    isLoading: modelsAreLoading(state),
+    modelsMessage: modelsMessage(state),
+    hasError: modelsHaveError(state),
+    hasData: modelsHaveData(state),
```

Comment:
you can add `metadata` here, since its in the redux state, and it will be available in `callOnMount`